### PR TITLE
Add failing tests

### PR DESCRIPTION
**Pull request by VincentLanglet**
This is all the error I get with this tool.

### CODE REVIEW

File: tests/EndToEnd/Bugs/errors.twig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{% if foo is defined %}{{ foo }}{% endif %}
```
Comment (VincentLanglet) [inline tests/EndToEnd/Bugs/errors.twig:1]: A defined check should prevent an "undefined variable" error.

File: tests/EndToEnd/Bugs/errors.twig [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+{% set languagesTo = ''|split(',')|filter(v => v != null) %}
```
Comment (VincentLanglet) [inline tests/EndToEnd/Bugs/errors.twig:3]: The closure/arrow function seems to have issues

File: tests/EndToEnd/Bugs/errors.twig [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+{% macro _display(data) %}
```
Comment (VincentLanglet) [inline tests/EndToEnd/Bugs/errors.twig:5]: Macro should at least be ignored

File: tests/EndToEnd/RenderPoints/render.html.twig [hunk 4]
```diff
@@ -5,3 +5,18 @@
 {% endfor %}
 
 {{ error }}
+
+{#
+    Twigstan compile this to if (isset($optional) && !(\null === $optional)) {
+    instead of if (array_key_exists($optional, $context) && !(\null === $optional)) {
+#}
+{% if optional is defined and optional is not null %}
```
Comment (VincentLanglet) [inline tests/EndToEnd/RenderPoints/render.html.twig:13]: There is false positive error here

File: tests/EndToEnd/RenderPoints/render.html.twig [hunk 5]
```diff
@@ -5,3 +5,18 @@
...
+{% endif %}
+
+{% for artist in artists %}
+    {% for artist2 in artists %} {# The same _parent variable is used for the two for loop #}
+        {% set foo = artist %}
+    {% endfor %}
+
+    {{ artist }}
+{% endfor %}
```
Comment (VincentLanglet) [inline tests/EndToEnd/RenderPoints/render.html.twig:22]: This code generate an `undefined _parent` error.

File: tests/EndToEnd/RenderPoints/render.html.twig (hunk 4 above)
Comment (ruudk) [reply tests/EndToEnd/RenderPoints/render.html.twig:13]: I found the problem. It was interesting to solve this. 
Fixed in https://github.com/twigstan/twigstan/pull/40

File: tests/EndToEnd/Bugs/errors.twig (hunk 1 above)
Comment (ruudk) [reply tests/EndToEnd/Bugs/errors.twig:1]: I think this is solved by https://github.com/twigstan/twigstan/pull/40

File: tests/EndToEnd/RenderPoints/render.html.twig (hunk 5 above)
Comment (ruudk) [reply tests/EndToEnd/RenderPoints/render.html.twig:22]: Created a PR with the failing test here: https://github.com/twigstan/twigstan/pull/35

Not yet fixed.

File: tests/EndToEnd/Bugs/errors.twig (hunk 3 above)
Comment (ruudk) [reply tests/EndToEnd/Bugs/errors.twig:5]: See #4

File: tests/EndToEnd/Bugs/errors.twig (hunk 2 above)
Comment (ruudk) [reply tests/EndToEnd/Bugs/errors.twig:3]: Created a PR with the failing test here #36 

Not yet solved

**Comment by ruudk** [comment]
Closing this PR as it's not tracked elsewhere.

**Pull request closed without merging**